packages/melos/lib/src/scripts.dart

```diff
@@ -129,15 +130,18 @@ class Script {
     required String workspacePath,
   }) {
     final scriptPath = 'scripts/$name';
-    String run;
+    var run = '';
     String? description;
     var env = <String, String>{};
+    var steps = <String>[];
```

Comment:
This could be final, and the steps could just be added in the `if` further down.
You probably just follow how `env` is done, but that one should probably also change to be final (not in necessarily baked into this PR though).

Reply:
if `steps` become an empty final variable declaration and then added further down in the `if` statement, it will actually cause the following error:

```
The final variable 'steps' can't be read because it's potentially unassigned at this point. (Documentation)  Ensure that it is assigned on necessary execution paths.
```

I tested some approaches, like the following, for example:
```
if (yaml is String) {
  run = yaml;
}

if (yaml is! Map<Object?, Object?>) {
  throw MelosConfigException('Unsupported value for script $name');
}

[...]
```
This way we will reverse the logic and expected positive result at the end without if statements, and a final variable for `steps` or `env` will work. But I'm missing something because in theoric should work but it's breaking everything ahahah

Reply:
It would need an `else` case too, then the compiler will know that it can be final and assigned.

Reply:
> But I'm missing something because in theoric should work but it's breaking everything ahahah

Hahah classic! I'm at my phone at the moment, but I can have a look tomorrow if you haven't solved it, I bet you will though. 😄

Reply:
Hey @spydon, I was able to figured it out! I like the first approach I just shared with you because we're going to get rid of the nested if statements and apply a sort of 'early return pattern', which was also a big help for me in order to  understand what the `Script.fromYaml` factory was trying to achieve.

In the commit e5f44b20939d99910c8815d0691ef398fc433ad3, I introduce this new approach and additionally, I made the `steps` variable a `final` one since it is assigned a value and doesn't change afterward. Let me know what you think.